gcsio/src/main/java/com/google/cloud/hadoop/gcsio/BatchHelper.java

```diff
@@ -70,6 +70,8 @@ private BatchHelper(HttpRequestInitializer requestInitializer, Storage gcs,
       long maxRequestsPerBatch) {
     this.pendingBatchEntries = new LinkedList<>();
     this.batch = gcs.batch(requestInitializer);
+    // Force use of API-specific batch endpoint
+    this.batch.getBatchUrl().appendRawPath("/" + Storage.DEFAULT_SERVICE_PATH);
```

Comment:
Is there anything that can / should be done to guard this from happening in 1.7?

Reply:
This change is intended only for 1.6.x branch, it will not be ported to master/1.7 branch, so there no need to guard it in any way.

Reply:
Added if condition, in case somebody will use 1.6 GCS connector with newer GCS API libs.